File: cpp_sim/188623/Zolw.cpp
#include "Zolw.h"

Zolw::Zolw(Swiat* swiatOrganizmu, int posX, int posY, int age, bool czyCzeka)
: Zwierze(swiatOrganizmu, 2, 1, posX, posY, age, czyCzeka){
    
};

void Zolw::rysowanie(){
    printf("Z");
};

string Zolw::getNazwa(){
    return ("Zolw");
};

void Zolw::akcja(){
    wiek++;
    // Szansa na poruszenie sie 25%
    int randomTick = rand() % 100 + 1;
    if(randomTick > 75){
        Zwierze::akcja();
    }
};

bool Zolw::odbicie(Organizm *entity){
    return (entity->getSila() < 5);
};

Organizm *Zolw::klonowanie(int posX, int posY){
    return new Zolw(swiatOrganizmu, posX, posY, 1, true);
};

Zolw::~Zolw(){
    
};

File: java_sim/simulation_188623/src/Zolw.java
import java.util.Random;

import javax.swing.ImageIcon;

public class Zolw extends Zwierze {

    Zolw(Swiat swiatOrganizmu, int x_pos, int y_pos, int wiek, boolean czyCzeka) {
        super(swiatOrganizmu, 2, 1, x_pos, y_pos, wiek, czyCzeka);
        this.ikonka = new ImageIcon("zdjecia/zolw.png");
        this.hexIkonka = new ImageIcon("zdjecia/zolwHex.png");
    }

    String getNazwa() {
        return ("Zolw");
    }

    @Override
    char getZnaczek() {
        return 'Z'; 
    }
    
    void akcja() {
        //szansa na poruszenie sie 25%
        Random rand = new Random();
        int randomTick = rand.nextInt(101);
        if(randomTick>75) {
            super.akcja();
        }
    }

    boolean odbicie(Organizm entity) {
        return (entity.getSila() < 5);
    }

    Organizm klonowanie(int posX, int posY) {
        return new Zolw(swiatOrganizmu, posX, posY, 1, true);
    }

}
